pkg/apply/applier.go

```diff
@@ -156,6 +159,46 @@ func (a *Applier) Run(ctx context.Context, invInfo inventory.InventoryInfo, obje
 		}
 		klog.V(4).Infof("calculated %d apply objs; %d prune objs", len(applyObjs), len(pruneObjs))
 
+		// Build list of apply sets, based on dependency ordering.
+		applySets, err := graph.SortObjs(applyObjs)
+		if err != nil {
+			validationErrors = append(validationErrors, err)
+		}
+		klog.V(4).Infof("apply sets: %d", len(applySets))
+		// Build list of prune sets, based on reverse dependency ordering.
+		pruneSets, err := graph.ReverseSortObjs(pruneObjs)
+		if err != nil {
+			validationErrors = append(validationErrors, err)
+		}
+		klog.V(4).Infof("prune sets: %d", len(pruneSets))
+		// Handle validation events
+		validationErrors = multierror.Unwrap(validationErrors...)
+		klog.V(4).Infof("validation errors: %d", len(validationErrors))
+		if len(validationErrors) > 0 {
+			for _, err := range validationErrors {
+				handleValidationError(eventChannel, err)
+			}
+			// Exit early, if configured to do so
+			if options.ValidationPolicy == validation.ExitEarly {
+				handleError(eventChannel, multierror.New(validationErrors))
+				return
+			}
+		}
+		// Extract unique IDs from ValidationErrors
+		invalidIds := invalidIds(validationErrors)
+		klog.V(4).Infof("invalid objects: %d", len(invalidIds))
+		// Don't apply or prune invalid objects
+		// TODO: what about dependencies?
```

Comment:
This will break waiting for dependencies if the dependency is invalid, but only if SkipInvalid is enabled. Since ExitEarly is default for reverse compatibility, this is not technically a regression. The intended fix will be a new feature to filter objects whose dependencies aren't reconciled. The dependency filter is discussed in https://github.com/kubernetes-sigs/cli-utils/issues/455